test/ERC4626.t.sol

```diff
@@ -159,6 +166,116 @@ contract SparkVaultERC4626Test is SparkVaultTestBase {
+        assertEq(maxDepositUser1, 0);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6);
+        maxDepositUser1 = vault.maxDeposit(user1);
+        maxDepositUser2 = vault.maxDeposit(user2);
+        assertEq(maxDepositUser1, 0);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 + 1);
+        maxDepositUser1 = vault.maxDeposit(user1);
+        maxDepositUser2 = vault.maxDeposit(user2);
+        assertEq(maxDepositUser1, 1);
+        assertEq(maxDepositUser2, maxDepositUser1);
+    }
+
+    function test_maxDeposit_depositCapGtTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        uint256 maxDepositUser1 = vault.maxDeposit(user1);
+        uint256 maxDepositUser2 = vault.maxDeposit(user2);
+
+        assertEq(maxDepositUser1, 1_099_892.540218e6);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        assertEq(maxDepositUser1, vault.depositCap() - vault.totalAssets());
+    }
+
+    function test_maxMint_depositCapLeTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        assertEq(vault.totalAssets(), 1_000_107.459782e6);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 - 1);
+        uint256 maxMintUser1 = vault.maxMint(user1);
+        uint256 maxMintUser2 = vault.maxMint(user2);
+        assertEq(maxMintUser1, 0);
+        assertEq(maxMintUser2, maxMintUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6);
+        maxMintUser1 = vault.maxMint(user1);
+        maxMintUser2 = vault.maxMint(user2);
+        assertEq(maxMintUser1, 0);
+        assertEq(maxMintUser2, maxMintUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 + 1);
+        maxMintUser1 = vault.maxMint(user1);
+        maxMintUser2 = vault.maxMint(user2);
+        // Since shares are worth more than assets at this point, increasing depositCap (assets) by
+        // just 1 doesn't mean that shares jumps up immediately
+        assertEq(maxMintUser1, 0);
+        assertEq(maxMintUser2, maxMintUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 + 2);
+        maxMintUser1 = vault.maxMint(user1);
+        maxMintUser2 = vault.maxMint(user2);
+        // After increasing depositCap (assets) by 2, shares starts to reflect that
+        assertEq(maxMintUser1, 1);
+        assertEq(maxMintUser2, maxMintUser1);
+    }
+
+    function test_maxMint_depositCapGtTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        uint256 maxMintUser1 = vault.maxMint(user1);
+        uint256 maxMintUser2 = vault.maxMint(user2);
+
+        assertEq(maxMintUser1, 1_099_774.358705e6);
```

Comment:
Should derive this number, like (vault.depositCap - vault.totalAssets) * 1e27 / vault.nowChi()